### DISCUSSION TITLE
Can you log in with a password?

### DISCUSSION THREAD

**Discussion by Tomorrowxxy** [spatie/ssh#75] [Ideas]
Can you log in with a password?
I am using machine A to log in to B... N through the intranet, and I hope to log in through the password. At present, it is not supported. Is there a plan to support it?

**Comment by FlaviuRadulescu** [comment]
I have made it work doing this:

added 2 methods and changed the getExecuteCommand from the Ssh.php file:

        public function setPassword(string $password)
        : self {

            $this->password = "sshpass -p " . $password;

            return $this;
        }

        private function getPassword() {

            return $this->password ?? '';
        }

        /**
         * @param  string|array  $command
         *
         * @return string
         */
        public function getExecuteCommand($command)
        : string {

            $commands = $this->wrapArray($command);
            $extraOptions = implode(' ', $this->getExtraOptions());
            $commandString = implode(PHP_EOL, $commands);
            $delimiter = 'EOF-SPATIE-SSH';
            $target = $this->getTargetForSsh();
            $password = $this->getPassword();
            if (in_array($this->host, ['local', 'localhost', '127.0.0.1'])) {
                return $commandString;
            }

            return "{$password} ssh {$extraOptions} {$target} 'bash -se' << \\$delimiter" . PHP_EOL . $commandString . PHP_EOL . $delimiter;
        }

you will need to install the sshpass tool into your OS (for centos you could do: `yum install sshpass`)

Hope it helps somebody and maybe will be integrated into the main code

**Reply by mehran-mrn** [reply]
not work for me.
